source/elements/oneDAL/source/data_management/index.rst

```diff
@@ -5,12 +5,294 @@
+  implementation details, which (1) are unnecessary for user, (2) allow to
+  select most appropriate table implementation for every particular case.
+
+- Providing an additional information how to create a :ref:`table` inside an
+  algorithm for :ref:`results  <Result>`. This information may include metadata,
+  memory allocators need to be used, or even particular implementation of a
+  table.
+
+For details, see :ref:`table-builders` section.
+
+.. _accessor:
+
+Accessor
+--------
+
+Accessor is a concept which defines sinle way how to get the data from a
+in-memory numerical :ref:`dataset`. It allows:
+
+- To make uniform access to the data from various sets of different objects,
+  like :ref:`tables <Table>` or :ref:`table builders <table-builder>` without
+  exposing their implementation details.
+
+- To convert a variety of numeric :term:`data formats <Data format>` into a
+  smaller set (e.g. for better vectorization).
+
+- To give an ability to focus on the data acquisition in the desired :term:`data
+  format` with a wanted set of operations over the data.
+
+- To make read-only, read-write and write-only access to the data.
+
+For details, see :ref:`accessors` section.
+
+Use-case example for table, accessor and table builder
+------------------------------------------------------
+
+This section gives an basic usage scenario of :ref:`table`, :ref:`table-builder`
+and :ref:`accessor` concepts and relations between them. The following diagram
+shows objects of these concepts, which are highlighted by colors:
+
+- :ref:`table-builder` objects are blue
+
+- :ref:`table` objects are cyan
+
+- :ref:`accessors <Accessor>` are yellow
+
+- objects with gray background color are not a part of |dal_short_name|
+  specification and they are provided just for illustration purposes
+
+To perform some computations based on the dataset, one shall create a
+:ref:`table` object first. It can be done using :ref:`data-source` or
+:ref:`table-builder` objects depending on the situation. The diagram briefly
+shows a situation when :ref:`table` is created from :ref:`table-builder`
+interatively with calls of various member functions that update internal state
+of the builder. Then, the `build()` method is called which takes a snapshot of
+current builder state and creates a table from it.
+
+Once table object created, the data inside it can be accessed using member
+functions of table object or with help of read-only accessor as shown on the
+diagram. Table can be used in the computations as input or parameter of some
+algorithm.
+
+Algorithms results also contain table objects. If one needs to change the data
+inside some table, builder object can be constructed for this. If table object
+is unique (just only one reference exists), it can be moved inside builder -
+thus no copy operations of data will be performed. Data inside table builder can
+be retrieved by read-only, write-only or read-write accessors.
```

Comment:
```suggestion
Algorithms' results also contain table objects. If one needs to change the data
within some table, a builder object can be constructed for this. If a table object
is unique (meaning that only one reference exists), it can be moved inside a builder.
Thus, no copy operations of data will be performed. Data inside a table builder can
be retrieved by read-only, write-only or read-write accessors.
```